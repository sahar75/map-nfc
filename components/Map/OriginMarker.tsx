import React, { useMemo } from "react";
import { Marker } from "react-native-maps";
const locationImg = require("../../assets/images/origin.png");

interface IProps {
  selectedRoutePoints: {
    lat: number;
    lng: number;
  }[];
}

const OriginMarker: React.FC<IProps> = ({ selectedRoutePoints }) => {
  const origin = useMemo(() => selectedRoutePoints?.[0], [selectedRoutePoints]);

  return (
    <Marker
      coordinate={{ latitude: origin?.lat ?? 0, longitude: origin?.lng ?? 0 }}
      title="Origin"
      image={locationImg}
      anchor={{ x: 0.45, y: 0.7 }}
    />
  );
};

export default React.memo(OriginMarker);
