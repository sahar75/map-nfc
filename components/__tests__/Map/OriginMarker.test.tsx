// OriginMarker.test.tsx
import React from 'react';
import TestRenderer from 'react-test-renderer';
import OriginMarker from '../../Map/OriginMarker';

// Mock the Marker component from react-native-maps
jest.mock('react-native-maps', () => {
  return {
    Marker: 'Marker',
  };
});

describe('OriginMarker', () => {
  it('should render correctly with given route points', () => {
    const selectedRoutePoints = [{ lat: 37.78825, lng: -122.4324 }];
    const tree = TestRenderer.create(
      <OriginMarker selectedRoutePoints={selectedRoutePoints} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with no route points', () => {
    const selectedRoutePoints: { lat: number; lng: number }[] = [];
    const tree = TestRenderer.create(
      <OriginMarker selectedRoutePoints={selectedRoutePoints} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
