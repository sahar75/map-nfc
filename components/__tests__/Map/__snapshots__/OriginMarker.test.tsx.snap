// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OriginMarker should render correctly with given route points 1`] = `
<Marker
  anchor={
    {
      "x": 0.45,
      "y": 0.7,
    }
  }
  coordinate={
    {
      "latitude": 37.78825,
      "longitude": -122.4324,
    }
  }
  image={
    {
      "testUri": "../../../assets/images/origin.png",
    }
  }
  title="Origin"
/>
`;

exports[`OriginMarker should render correctly with no route points 1`] = `
<Marker
  anchor={
    {
      "x": 0.45,
      "y": 0.7,
    }
  }
  coordinate={
    {
      "latitude": 0,
      "longitude": 0,
    }
  }
  image={
    {
      "testUri": "../../../assets/images/origin.png",
    }
  }
  title="Origin"
/>
`;
